WrightTools/kit/_array.py

```diff
@@ -237,7 +237,15 @@ def share_nans(*arrs) -> tuple:
     list
         List of nD arrays in same order as given, with nan indicies syncronized.
     """
-    nans = np.zeros(joint_shape(*arrs))
+    typ = int
+    for arr in arrs:
+        if arr.dtype.kind in "fmM":
+            if typ != np.complex128:
+                typ = arr.dtype
+        if arr.dtype.kind in "c":
+            typ = arr.dtype
+
+    nans = np.zeros(joint_shape(*arrs), dtype=typ)
     for arr in arrs:
         nans *= arr
```

Comment:
Note that `np.nan * complex(0) = nan + nanj`, so no concern here about 0-valued imaginary parts .  This step should work